airbyte-workers/src/main/java/io/airbyte/workers/temporal/TemporalUtils.java

```diff
@@ -37,59 +37,116 @@
-  private static final String HUMAN_READABLE_WORKFLOW_EXECUTION_TTL =
-      DurationFormatUtils.formatDurationWords(WORKFLOW_EXECUTION_TTL.toMillis(), true, true);
+  public static WorkflowServiceStubs createTemporalCloudService() {
+    return createTemporalCloudService(
+        configs.getTemporalCloudClientCert(),
+        configs.getTemporalCloudClientKey(),
+        configs.getTemporalCloudHost(),
+        configs.getTemporalCloudNamespace());
+  }
+
+  @VisibleForTesting
+  public static WorkflowServiceStubs createTemporalCloudService(
+                                                                final String temporalCloudClientCert,
+                                                                final String temporalCloudClientKey,
+                                                                final String temporalHost,
+                                                                final String temporalNamespace) {
+    try {
+      final InputStream clientCert = new ByteArrayInputStream(temporalCloudClientCert.getBytes(StandardCharsets.UTF_8));
+      final InputStream clientKey = new ByteArrayInputStream(temporalCloudClientKey.getBytes(StandardCharsets.UTF_8));
+
+      final WorkflowServiceStubsOptions options = WorkflowServiceStubsOptions.newBuilder()
+          .setSslContext(SimpleSslContextBuilder.forPKCS8(clientCert, clientKey).build())
+          .setTarget(temporalHost)
+          .build();
+
+      return getTemporalClientWhenConnected(
+          WAIT_INTERVAL,
+          MAX_TIME_TO_CONNECT,
+          WAIT_TIME_AFTER_CONNECT,
+          () -> WorkflowServiceStubs.newInstance(options),
+          temporalNamespace);
+    } catch (final Exception e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  public static WorkflowServiceStubs createTemporalAirbyteService() {
+    return createTemporalAirbyteService(configs.getTemporalHost());
+  }
+
+  @VisibleForTesting
+  public static WorkflowServiceStubs createTemporalAirbyteService(final String temporalHost) {
+    final WorkflowServiceStubsOptions options = WorkflowServiceStubsOptions.newBuilder()
+        .setTarget(temporalHost)
+        .build();
+
+    final WorkflowServiceStubs temporalService = getTemporalClientWhenConnected(
+        WAIT_INTERVAL,
+        MAX_TIME_TO_CONNECT,
+        WAIT_TIME_AFTER_CONNECT,
+        () -> WorkflowServiceStubs.newInstance(options),
+        DEFAULT_NAMESPACE);
+
+    configureTemporalAirbyteNamespace(temporalService);
+
+    return temporalService;
+  }
 
-  public static void configureTemporalNamespace(final WorkflowServiceStubs temporalService) {
+  public static String getProductionNamespace() {
+    return configs.temporalCloudEnabled() ? configs.getTemporalCloudNamespace() : TemporalUtils.DEFAULT_NAMESPACE;
+  }
+
+  private static void configureTemporalAirbyteNamespace(final WorkflowServiceStubs temporalService) {
```

Comment:
Reworked this to only happen when instantiating an airbyte-temporal service. It doesn't make sense for us to configure Temporal Cloud (and we get unauthorized exceptions if we try)